internal/provider/resource_gitlab_deploy_key_enable_test.go

```diff
@@ -2,145 +2,129 @@ package provider
 
 import (
 	"fmt"
-	"strconv"
-	"strings"
 	"testing"
 
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
-	gitlab "github.com/xanzy/go-gitlab"
+	"github.com/xanzy/go-gitlab"
 )
 
 func TestAccGitlabDeployKeyEnable_basic(t *testing.T) {
-	var deployKey gitlab.ProjectDeployKey
-	rInt := acctest.RandInt()
+	testAccCheck(t)
+
+	testProjectParent := testAccCreateProject(t)
+	testProjectKeyShared := testAccCreateProject(t)
 
-	keyTitle := "main"
 	key := "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDblguSWgpqiXIjHPSas4+N3Dten7MTLJMlGQXxGpaqN9nGPdNmuRB2YXyjT/nrryoY/qrtuVkPnis5WVo8N/s3hAnJbeJPUS2WKEGjpBlL34AQ+ANnlmGY8L6zr82Hp2Ommb7XGGtlq5D3yLCgTfcXLjC51tgcdwHsdH1U+RisgLwaTSrP/HF4G7IAr5ATsyYjtCwQRQ8ijdf5A34+XN6h8J6TLXKab5eZDuH38s9LxJuS7MRxx/P2UTOsqfjtrZWoQgE5adEGvnDxKyruex9PzNbCNVahzsma7tdikDbzxlHLIZ1aht6rKuai3iyLgcZfGIYtkq4xvg/bnNXxSsGf worker@kg.getwifi.com"
 
+	canPushDeployKeyOptions := gitlab.AddDeployKeyOptions{
+		Title:   gitlab.String("main"),
+		Key:     gitlab.String(key),
+		CanPush: gitlab.Bool(true),
+	}
+
+	parentProjectDeployKey := testAccCreateDeployKey(t, testProjectParent.ID, &canPushDeployKeyOptions)
+
 	resource.Test(t, resource.TestCase{
 		PreCheck:          func() { testAccPreCheck(t) },
 		ProviderFactories: providerFactories,
 		CheckDestroy:      testAccCheckGitlabDeployKeyEnableDestroy,
 		Steps: []resource.TestStep{
-			// Create a project and deployKey with default options
+			// Enable a deployKey on project with default options
+			{
+				Config: testAccGitlabDeployKeyEnableConfig(testProjectKeyShared.ID, parentProjectDeployKey.ID),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet("gitlab_deploy_key_enable.foo", "key"),
+					resource.TestCheckResourceAttrSet("gitlab_deploy_key_enable.foo", "title"),
```

Comment:
Those are also non-computed fields. The thing is that the verify import test step will actually check that the imported state will match the config, which implies that these attributes are correctly set into state.

Reply:
e.g. for a little experiment: try to remove these checks here and also remove the line `105` in `internal/provider/resource_gitlab_deploy_key.go` where the `title` is set to state. The subsequent `Verify import` step at line 42 here should catch that :)

Reply:
also don't hesitate to ping me on Discord if you want to discuss this in detail :)